docs/release.md

```diff
@@ -0,0 +1,138 @@
+# Rancher Turtles Releases
+
+## Release Cadence
+
+- Rancher Turtles minor versions (v0.**4**.0 versus v0.**3**.0) are released every 2-3 weeks or once a month.
+- Rancher Turtles patch versions (v0.4.**2** versus v0.4.**1**) are released as often as weekly. 
+
+## Release Process
+
+1. Clone the repository locally: 
+
+```bash
+git clone git@github.com:rancher-sandbox/rancher-turtles.git
+```
+
+2. Depending on whether you are cutting a minor or patch release, the process varies.
+
+    * If you are cutting a new minor release:
+
+        Create a new release branch (i.e release-X) and push it to the upstream repository.
+
+        ```bash
+            # Note: `upstream` must be the remote pointing to `github.com/rancher-sandbox/rancher-turtles`.
+            git checkout -b release-0.4
+            git push -u upstream release-0.4
+            # Export the tag of the minor release to be cut, e.g.:
+            export RELEASE_TAG=v0.4.0
+        ```
+    * If you are cutting a patch release from an existing release branch:
+
+        Use existing release branch.
+
+        ```bash
+            # Note: `upstream` must be the remote pointing to `github.com/rancher-sandbox/rancher-turtles`
+            git checkout upstream/release-0.4
+            # Export the tag of the patch release to be cut, e.g.:
+            export RELEASE_TAG=v0.4.1
+        ```
+3. Create a signed/annotated tag and push it:
+
+```bash
+# Create tags locally
+git tag -s -a ${RELEASE_TAG} -m ${RELEASE_TAG}
+
+# Push tags
+git push upstream ${RELEASE_TAG}
+```
+
+This will trigger a [release GitHub action](https://github.com/rancher-sandbox/rancher-turtles/blob/main/.github/workflows/release.yaml) that creates a release with Rancher Turtles components.
+If you are cutting a new minor/major release, please follow the [next step](#post-release-steps-in-rancher-turtles-docs) below, otherwise skip.
+
+## Post-release steps in Rancher Turtles Docs
+
+If a new minor or major branch was created, there are some post-release actions that need to be taken.
+
+### Create a new tag for Rancher Turtles Docs repo
+
+1. Clone the Rancher Turtles Docs repository locally: 
+
+```bash
+git clone git@github.com:rancher-sandbox/rancher-turtles-docs.git
+```
+
+2. Export the tag of the minor/major release, create a signed/annotated tag and push it:
+
+```bash
+# Export the tag of the minor/major release in a format of v0.X/v1.X, e.g.:
+export RELEASE_TAG=v0.4
+
+# Create tags locally
+git tag -s -a ${RELEASE_TAG} -m ${RELEASE_TAG}
+
+# Push tags
+# Note: `upstream` must be the remote pointing to `github.com/rancher-sandbox/rancher-turtles-docs`
+git push upstream ${RELEASE_TAG}
+```
+
+### Create a patch for new version of Rancher Turtles Docs
```

Comment:
This step is automatic, but this requires to adjust permissions on the repo temporarily here - https://github.com/rancher-sandbox/rancher-turtles-docs/settings/rules/124836. It pushes the version change on behalf of the tag committer.